src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1138,4 +1138,15 @@ protected static function getMutatorMethods($class)
 
         return $matches[1];
     }
+    
+    /**
+     * Checks if a specific attribute exists.
+     *
+     * @param string $key
+     * @return bool
+     */
+    protected function hasAttribute($key)
+    {
+        return array_key_exists($key, $this->attributes);
```

Comment:
Will this work on mutated attributes? You should probably be using attributesToArray() instead of the attributes property.

Reply:
I am not sure on this.. because this method only checks if this key (i.e., the field) is set.. The value is not needed for this method..

As far as i know, mutators should only work on `attributes` (e.g., apply additional functions on the values in order to change formatting or whatever). Therefore, `$this->attributes` should be enough for the scope of this function.. 

Or am I missing something really critical?!

Reply:
Not quite true, although mutators should have a relationship with attributes someone could create a totally independent mutator.

But what I meant in the first place is, no mater if the attribute is a mutator or not this should return true, if not it could lead to confusion as I do not need to know if the attribute was a mutator in the first place or a real attribute.

Reply:
i changed it to `attributesToArray()` - thank you for pointing that out..